browser/browsing_data/brave_clear_browsing_data_browsertest.cc

```diff
@@ -0,0 +1,383 @@
+};
+
+IN_PROC_BROWSER_TEST_F(BraveClearDataOnExitTest, NoPrefsSet) {
+  // No set preferences to clear data.
+  SetExepectedRemoveDataCallCount(0);
+  // Tell the application to quit.
+  chrome::ExecuteCommand(browser(), IDC_EXIT);
+}
+
+IN_PROC_BROWSER_TEST_F(BraveClearDataOnExitTest, VerifyRemovalMasks) {
+  // Set all clear data on exit preferences and corresponding expected remove
+  // mask and origin flags.
+  SetClearAll(browser()->profile()->GetPrefs());
+
+  // Given those preferences the following removal mask is expected.
+  SetExpectedRemoveDataRemovalMasks(GetRemoveMaskAll(), GetOriginMaskAll());
+
+  // Expect a call to clear data.
+  SetExepectedRemoveDataCallCount(1);
+
+  // Tell the application to quit.
+  chrome::ExecuteCommand(browser(), IDC_EXIT);
+}
+
+class BraveClearDataOnExitTwoBrowsersTest : public BraveClearDataOnExitTest {
+ public:
+  BraveClearDataOnExitTwoBrowsersTest() : BraveClearDataOnExitTest() {
+    browsers_count_ = 2u;
+  }
+
+ protected:
+  // Open a new browser window with the provided |profile|.
+  Browser* NewBrowserWindow(Profile* profile) {
+    DCHECK(profile);
+    ui_test_utils::BrowserAddedObserver browser_added_observer;
+    chrome::NewEmptyWindow(profile);
+    Browser* browser = browser_added_observer.WaitForSingleNewBrowser();
+    DCHECK(browser);
+    content::WaitForLoadStopWithoutSuccessCheck(
+      browser->tab_strip_model()->GetActiveWebContents());
+    return browser;
+  }
+
+  // Open a new browser window with a guest session.
+  Browser* NewGuestBrowserWindow() {
+    ui_test_utils::BrowserAddedObserver browser_added_observer;
+    profiles::SwitchToGuestProfile(ProfileManager::CreateCallback());
+    Browser* browser = browser_added_observer.WaitForSingleNewBrowser();
+    DCHECK(browser);
+    // When a guest |browser| closes a BrowsingDataRemover will be created and
+    // executed. It needs a loaded TemplateUrlService or else it hangs on to a
+    // CallbackList::Subscription forever.
+    Profile* guest = g_browser_process->profile_manager()->GetProfileByPath(
+        ProfileManager::GetGuestProfilePath());
+    DCHECK(guest);
+    search_test_utils::WaitForTemplateURLServiceToLoad(
+        TemplateURLServiceFactory::GetForProfile(guest));
+    // Navigate to about:blank.
+    ui_test_utils::NavigateToURL(browser, GURL(url::kAboutBlankURL));
+    content::WaitForLoadStopWithoutSuccessCheck(
+      browser->tab_strip_model()->GetActiveWebContents());
+    return browser;
+  }
+
+  // Open a new browser window with a new profile.
+  Browser* NewProfileBrowserWindow() {
+    base::FilePath path;
+    base::PathService::Get(chrome::DIR_USER_DATA, &path);
+    path = path.AppendASCII("Profile 2");
+
+    base::ScopedAllowBlockingForTesting allow_blocking;
+    ProfileManager* profile_manager = g_browser_process->profile_manager();
+    size_t starting_number_of_profiles = profile_manager->GetNumberOfProfiles();
+    if (!base::PathExists(path) && !base::CreateDirectory(path))
```

Comment:
Is it cleaned up somewhere?
I'd suggest to just rewrite this using `base::ScopedTempDir`

Reply:
Done.